core/services/gateway/gateway.go

```diff
@@ -66,6 +67,7 @@ func NewGateway(codec Codec, httpServer gw_net.HttpServer, handlers map[string]H
 	}
 	// Connect servers to the Gateway
 	httpServer.SetHTTPRequestHandler(gw)
+	wsServer.SetConnectionAcceptor(connMgr)
```

Comment:
Looks like Gateway does not really need wsServer. If ConnectionManager owns it, let it creates it and controls its lifecycle. In this case you won't need this `SetConnectionAcceptor`, and an extra dependency for Gateway.